lib/spack/spack/modules/lmod.py

```diff
@@ -319,9 +318,10 @@ def unlocked_paths(self):
                 hierarchy = self.conf.hierarchy_tokens
                 available = self.conf.available
                 token2path = lambda x: self.token_to_path(x, available[x])
-                l = [x for x in hierarchy if x in item]
                 parts = []
-                for x in l:
+                for x in hierarchy:
```

Comment:
I find the lines below slightly less readable without a list comprehension.

Reply:
I guess I like the fact that my version only iterates through the data once.

Reply:
Optimizing is good, but I don't think people will use more than three / four items in the hierarchy :smile: